### PR TITLE
Fix deprecated API usage for embedded fonts on iOS/Catalyst 18.0+

### PR DESCRIPTION
**Pull request by drasticactions**
### Description of Change

`CTFontManager.RegisterGraphicsFont` throws a [deprecation warning](https://developer.apple.com/documentation/coretext/1499499-ctfontmanagerregistergraphicsfon) for iOS 18+. This was breaking my build.

https://github.com/xamarin/xamarin-macios/blob/main/src/CoreText/CTFontManager.cs#L445-L462

This PR attempts to change it to a supported API for these platforms, using `CTFontManager.RegisterFontsForUrl`. To do this, I believe, we need to write the font to a file on disk (in this case, in temp storage in the users application cache) and then register it.

### CODE REVIEW

File: src/Core/src/Fonts/EmbeddedFontLoader.iOS.cs
Comment (rmarinho) [inline src/Core/src/Fonts/EmbeddedFontLoader.iOS.cs:62]: can we use is null ? and drop the private from the methods

**Comment by rmarinho** [comment]
I think this should target to net9 when we merge the #25017

**Comment by rmarinho** [comment]
Net9 is on iOS18 XCODE16 can we target that branch ?

**Comment by PureWeen** [comment]
> Net9 is on iOS18 XCODE16 can we target that branch ?

I think this is fine for main so that if anyone is on XCODE 16 locally they are still able to compile main until we merge net9.0 into main

**Review by PureWeen — CHANGES_REQUESTED** [review]
Don't merge until 
https://github.com/dotnet/maui/pull/25055

is green and merged

**Comment by PureWeen** [comment]
Because we merged
https://github.com/dotnet/maui/pull/25092 to fix the compile error on main

I'm going to close this PR in favor of
https://github.com/dotnet/maui/pull/25055

**Pull request closed without merging**
